actions/drain_node_handler.go

```diff
@@ -202,10 +239,26 @@ func (h *drainNodeHandler) listNodePodsToEvict(ctx context.Context, node *v1.Nod
 		return nil, fmt.Errorf("listing node %v pods: %w", node.Name, err)
 	}
 
-	podsToEvict := lo.Filter(pods.Items, func(pod v1.Pod, _ int) bool {
-		return !isDaemonSetPod(&pod) && !isStaticPod(&pod)
-	})
+	podsToEvict := make([]v1.Pod, 0)
+	castPods := make([]v1.Pod, 0)
+	// Evict CAST PODs as last ones
+	for _, p := range pods.Items {
+		// Skip pods that have been recently removed.
+		if !p.ObjectMeta.DeletionTimestamp.IsZero() &&
+			int(time.Since(p.ObjectMeta.GetDeletionTimestamp().Time).Seconds()) > h.cfg.skipDeletedTimeoutSeconds {
+			continue
+		}
+
+		if p.Namespace == h.cfg.castNamespace && !isDaemonSetPod(&p) && !isStaticPod(&p) {
+			castPods = append(castPods, p)
+		}
 
+		if !isDaemonSetPod(&p) && !isStaticPod(&p) {
```

Comment:
I've read somewhere that Completed pods fail eviction. We should just skip those if they are part of the list.

Reply:
I was checking this by scheduling jobs in my cluster and eviction was working. But I can add the rule for it.

Reply:
No need to send evict calls for Completed pods. This will lower the API calls count in some scenarios.

Reply:
Added